pkg/strc/middleware.go

```diff
@@ -35,31 +35,4 @@ type MiddlewareConfig struct {
 
 	// ServerErrorLevel is the log level for requests with server errors (5xx). Defaults to Error.
 	ServerErrorLevel slog.Level
-
-	// SpanName is the name of the span. Defaults to "http request".
-	SpanName string
-
-	// WithUserAgent enables logging of the User-Agent header. Defaults to false.
-	WithUserAgent bool
-
-	// WithRequestBody enables logging of the request body. Defaults to false.
-	WithRequestBody bool
-
-	// WithRequestHeader enables logging of the request headers. Defaults to false.
-	WithRequestHeader bool
-
-	// WithResponseBody enables logging of the response body. Defaults to false.
-	WithResponseBody bool
-
-	// WithResponseHeader enables logging of the response headers. Defaults to false.
-	WithResponseHeader bool
-
-	// WithSpanID enables logging of the span ID. Defaults to false.
-	WithTraceID bool
-
-	// WithSpanID enables logging of the span ID. Defaults to false.
-	WithSpanID bool
-
-	// NoExtractTraceID disables extracting trace id from incoming requests. Defaults to false.
-	NoTraceContext bool
```

Comment:
I would also drop the body reader/writer implementation, I know it is needed for body length logging but I would rather drop that attribute for the sake of code simplification.

Reply:
Because for tracing the fully work, we'd still need to do tracing enabled http clients, right?

Reply:
Ah yes, I haven't realized that.